### PR TITLE
Add `winslash = "/"` parameter

### LINKED ISSUE #315: Improve handling of non-PCRE compatible strings in `.Rbuildignore`
In 'codemetar' v0.3.1 (but not, I believe, earlier versions), `codemetar::write_codemeta()` throws errors when `.Rbuildignore` contains strings that are not formatted as perl-compatible regular expressions, for example `man-roxygen`.

R and CRAN do not complain about plain-text formatted strings in `.Rbuildignore`, so I assume that this is a legitimate syntax.  Even if not, it would be nice if a more user-friendly message were provided than the one copied below; it took me a while to get to the bottom of the issue.

```
Error in grepl(paste0("^", ignore_dirs, collapse = "|"), files, perl = TRUE) : 
  invalid regular expression '^C:\PATH\TO\MYPACKAGE/man-roxygen'
In addition: Warning message:
In grepl(paste0("^", ignore_dirs, collapse = "|"), files, perl = TRUE) :
  PCRE pattern compilation error
	'unrecognized character follows \'
	at 'MYPACKAGE/man-roxygen'
```

System information:
```
R version 4.1.0 Patched (2021-06-21 r80544)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_United Kingdom.1252  LC_CTYPE=English_United Kingdom.1252   
[3] LC_MONETARY=English_United Kingdom.1252 LC_NUMERIC=C                           
[5] LC_TIME=English_United Kingdom.1252    
```

### PR DESCRIPTION
**Pull request by ms609**
Resolves #315

### CODE REVIEW

**Comment by maelle** [comment]
Thanks a ton @ms609! Would you mind 

* adding a comment above the change explaining why this is needed/how this works?
* also updating the NEWS file (list the bug fix, the PR number, and your GitHub username)?

**Comment by ms609** [comment]
Done both the above – my rationale for the `normalizePath()` setting is "for consistency with the rest of the code", so the original author may be better placed to justify it!

**Comment by cboettig** [comment]
good catch, thanks!

File: R/guess_other_metadata.R [hunk 1]
```diff
@@ -72,7 +73,7 @@ guess_fileSize <- function(root = ".", .ignore = NULL) {
   files <- gsub(paste0(normalizePath(root, winslash = "/"), "/"), "", files)
 
   ## exclude full dirs matching an ignore pattern
-  dirs <- list.dirs(normalizePath(root))
+  dirs <- list.dirs(normalizePath(root, winslash = "/"))
```
Comment (cboettig) [inline R/guess_other_metadata.R:76]: :+1:

**Review by cboettig — APPROVED** [review]

**Pull request merged**